### PR TITLE
Return errors decoding a response cache entry.

### PR DESCRIPTION
**Pull request by spencerkimball**
This is the correct thing to do--warn the user if idempotence guarantee
fails. Also, this prevents us from possibly causing replica divergence.

### CODE REVIEW

File: storage/range.go [hunk 1]
```diff
@@ -732,16 +732,13 @@ func (r *Range) applyRaftCommand(index uint64, originNodeID multiraft.NodeID, ar
...
-			if ok { // this is a replay! extract error for return
-				log.V(1).Infof("found response cache entry for %+v", args.Header().CmdID)
-				return reply.Header().GoError()
-			}
-			// In this case there was an error reading from the response
-			// cache. Instead of failing the request just because we can't
-			// decode the reply in the response cache, we proceed as though
-			// idempotence has expired.
-			log.Errorf("unable to read result for %+v from the response cache: %s", args, err)
+		if ok, err := r.respCache.GetResponse(header.CmdID, reply); ok && err == nil {
```
Comment (vivekmenezes) [inline storage/range.go]: Can err != nil when ok == true?

Comment (spencerkimball) [inline storage/range.go]: Yes, in this particular case this PR deals with this happens. The "ok" bool is really maybe better understood as "found". ok && err != nil means the response cache entry is there but not decodable.

**Comment by tschottdorf** [comment]
LGTM

File: storage/range.go (hunk 1 above)
Comment (vivekmenezes) [inline storage/range.go]: Okay. Perhaps the error line you log later can be improved to reflect that possibility.

Either way LGTM

**Pull request merged**